.github/workflows/check-and-lint.yml

```diff
@@ -1,33 +1,37 @@
+name: basics
+
 on:
   workflow_call:
-
-name: basics
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
 
 jobs:
-  check:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - run: cargo check --all-targets
-
-  fmt:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - run: rustup toolchain install nightly --profile minimal
-      - run: rustup component add rustfmt --toolchain nightly
-      - run: cargo +nightly fmt --all -- --check
-
-  clippy:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - run: cargo clippy --all-targets -- -D warnings
-
-  sort:
+  basics:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        steps:
+          - name: check
+            run: cargo check --all-targets
+          - name: fmt
+            run: cargo +nightly fmt --all -- --check
+          - name: clippy
+            run: cargo clippy --all-targets -- -D warnings
+          - name: sort
+            run: |
+              cargo test -- --test-threads=1 --ignored
```

Comment:
is it necessary to run tests under the `sort` step?

maybe that could be a separate step

Reply:
Good catch! Yeah, that's not meant to be there